sklearn/metrics/tests/test_pairwise.py

```diff
@@ -57,24 +57,28 @@
 from sklearn.exceptions import DataConversionWarning
 
 
-def test_pairwise_distances():
+DTYPES = (np.float32, np.float64)
+
+
+@pytest.mark.parametrize("dtype", DTYPES)
+def test_pairwise_distances(dtype):
     # Test the pairwise_distance helper function.
     rng = np.random.RandomState(0)
 
     # Euclidean distance should be equivalent to calling the function.
-    X = rng.random_sample((5, 4))
+    X = rng.random_sample((5, 4)).astype(dtype)
     S = pairwise_distances(X, metric="euclidean")
     S2 = euclidean_distances(X)
     assert_array_almost_equal(S, S2)
 
     # Euclidean distance, with Y != X.
-    Y = rng.random_sample((2, 4))
+    Y = rng.random_sample((2, 4)).astype(dtype)
     S = pairwise_distances(X, Y, metric="euclidean")
     S2 = euclidean_distances(X, Y)
     assert_array_almost_equal(S, S2)
     # Check to ensure NaNs work with pairwise_distances.
-    X_masked = rng.random_sample((5, 4))
-    Y_masked = rng.random_sample((2, 4))
+    X_masked = rng.random_sample((5, 4)).astype(dtype)
+    Y_masked = rng.random_sample((2, 4)).astype(dtype)
     X_masked[0, 0] = np.nan
     Y_masked[0, 0] = np.nan
     S_masked = pairwise_distances(X_masked, Y_masked, metric="nan_euclidean")
```

Comment:
Let's check the dtype of S_masked and S2_masked below (I cannot do a suggestion on the lines below view github unfortunately).